src/js/game/systems/logic_gate.js

```diff
@@ -19,8 +18,7 @@ export class LogicGateSystem extends GameSystemWithFilter {
     }
 
     update() {
-        for (let i = 0; i < this.allEntities.length; ++i) {
-            const entity = this.allEntities[i];
+        this.allEntities.forEach(entity => {
```

Comment:
There is a reason why a for loop is used everywhere, and that is performance. `forEach` is simply too slow :)